toolbox/library/src/YarpRpc.cpp

```diff
@@ -0,0 +1,219 @@
+{
+    TriggerSignal = 0,
+};
+
+// BLOCK PIMPL
+// ===========
+
+class YarpRpc::impl
+{
+public:
+    std::string serverPortName;
+    std::string clientPortName;
+
+    std::unique_ptr<yarp::os::Network> network = nullptr;
+    yarp::os::RpcClient rpcClientPort;
+    yarp::os::Bottle rpcCommand;
+};
+
+// BLOCK CLASS
+// ===========
+
+YarpRpc::YarpRpc()
+    : pImpl{new impl()}
+{}
+
+YarpRpc::~YarpRpc() = default;
+
+unsigned YarpRpc::numberOfParameters()
+{
+    return Block::numberOfParameters() + 2;
+}
+
+bool YarpRpc::parseParameters(BlockInformation* blockInfo)
+{
+    const std::vector<ParameterMetadata> metadata{
+        {ParameterType::STRING, ParamIndex::PortName, 1, 1, "PortName"},
+        {ParameterType::STRING, ParamIndex::RpcCommand, 1, 1, "RpcCommand"}
+    };
+
+    for (const auto& md : metadata) {
+        if (!blockInfo->addParameterMetadata(md)) {
+            bfError << "Failed to store parameter metadata";
+            return false;
+        }
+    }
+
+    return blockInfo->parseParameters(m_parameters);
+}
+
+bool YarpRpc::configureSizeAndPorts(BlockInformation* blockInfo)
+{
+    // PARAMETERS
+    // ==========
+
+    if (!YarpRpc::parseParameters(blockInfo)) {
+        bfError << "Failed to parse parameters.";
+        return false;
+    }
+
+    // INPUT
+    // =====
+    //
+    // 1) Trigger signal
+    //
+    // OUTPUT
+    // ======
+    //
+    // No outputs
+    //
+
+
+    const bool ok = blockInfo->setPortsInfo(
+        {
+            // Inputs
+            {InputIndex::TriggerSignal, Port::Dimensions{1}, Port::DataType::DOUBLE},
+        },
+        {
+            // Outputs
+        });
+
+    if (!ok) {
+        bfError << "Failed to configure input / output ports.";
+        return false;
+    }
+
+    return true;
+}
+
+bool YarpRpc::initialize(BlockInformation* blockInfo)
+{
+    if (!Block::initialize(blockInfo)) {
+        return false;
+    }
+
+    using namespace yarp::os;
+
+    // PARAMETERS
+    // ==========
+
+    if (!YarpRpc::parseParameters(blockInfo)) {
+        bfError << "Failed to parse parameters.";
+        return false;
+    }
+
+    // Temp variable to store rpc command string
+    std::string command;
+
+    bool ok = true;
+    ok = ok && m_parameters.getParameter("PortName", pImpl->serverPortName);
+    ok = ok && m_parameters.getParameter("RpcCommand", command);
+
+    if (!ok) {
+        bfError << "Failed to read input parameters.";
+        return false;
+    }
+
+    // CLASS INITIALIZATION
+    // ====================
+
+    pImpl->network = std::make_unique<yarp::os::Network>();
+
+    if (!yarp::os::Network::initialized() || !yarp::os::Network::checkNetwork(5.0)) {
+        bfError << "YARP server wasn't found active.";
+        return false;
+    }
+
+    // Open a rpc client port
+    // TODO: Check if the client name can be improved
+    pImpl->clientPortName = "/rpcClient" + pImpl->serverPortName;
+
+    if (!pImpl->rpcClientPort.open(pImpl->clientPortName)) {
```

Comment:
I think you can open an anonymous port. I do not remember exactly how to do it, try with `...`

Reply:
Good call ;) it works with `...`